src/groovy/com/recomdata/extensions/ExtensionsRegistry.groovy

```diff
@@ -1,13 +1,74 @@
 package com.recomdata.extensions
 
+import groovy.transform.CompileStatic
+
 /**
  * Date: 14-Jan-16
  * Time: 18:08
  */
 class ExtensionsRegistry {
     def analysisTabExtensions = []
+    List<Map<String, Object>> tabs = new LinkedList<>()
+
+    ExtensionsRegistry() {
+    }
 
     void registerAnalysisTabExtension(Map<String, Object> config = [:], String extensionId, String resourcesUrl, String bootstrapFunction) {
         analysisTabExtensions.add([extensionId: extensionId, resourcesUrl: resourcesUrl, bootstrapFunction: bootstrapFunction, config: config])
     }
+
+    void registerTab(Map<String, Object> config = [:], String id, String title, String controller) {
+        def tab = [id: id, title: title, controller: controller, insertBefore: config.insertBefore]
+        insertBeforeFirstReference(tabs, tab)
```

Comment:
We can't just use `incorporateTab`, because we don't know order of calls to `registerTab`. And because extensions can use identifiers not only default tabs, but other plugin's tabs as well we can get following situation:
1. PluginA adds tabA, which has `insertBefore: 'tabB'`
2. PluginB adds tabB, which has `insertBefore: 'rwg'`

If we won't apply any reordering, then call to `incorporateTab` with `tabA` will place it at last position, because `tabB` not yet added to `tabs` collection. The subsequent call to `incorporateTab` with `tabB` will place it on position before `rwg` tab. 
So in result `tabs` collection we will have `['tabB', 'rwg', ..., 'tabA']`. Though, if in `registerTab` we insert `tabB` before `tabA` (because `tabA` has reference to `tabB` in `insertBefore` property), then we get correct result: `['tabA', 'tabB', 'rwg', ...]`

But I agree, the code of these methods is very similar and may be refactored using something like `findAndInsertAfter`.